src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -108,24 +90,12 @@ public PluginCompatTester(PluginCompatTesterConfig config) {
         runner = new ExternalMavenRunner(config.getExternalMaven());
     }
 
-    public PluginCompatReport testPlugins() throws IOException {
```

Comment:
The first layer of the PCT code invoked by `main`. By convention, this throws two types of exceptions up to `main`:

- Checked exceptions, i.e. subclasses of `PluginCompatibilityTesterException`, which represent a high-level compatibility failure, and which are always immediately fatal except when `-failFast` is specified (in which case we test all plugins and only throw at the end).
- Unchecked exceptions and errors, including `IOException` rethrown as `UncheckedIOException`, which are always immediately fatal even when using `-failFast false`. These should only happen when there is an infrastructure failure (not a compatibility failure) and are basically assertions.